File: apps/api/app/models/orbit.py
from beanie import Document, PydanticObjectId
from pydantic import BaseModel, Field


class OrbitStats(BaseModel):
    memberCount: int = 0
    postCount: int = 0


class Orbit(Document):
    slug: str
    name: str
    description: str | None = None
    iconUrl: str | None = None
    bannerUrl: str | None = None
    createdBy: PydanticObjectId | None = None
    stats: OrbitStats = Field(default_factory=OrbitStats)

    class Settings:
        name = "orbits"

File: packages/db/src/models/Orbit.ts
import { Schema, type InferSchemaType, type Types } from "mongoose";
import { getModel } from "../lib/model.js";

const OrbitSchema = new Schema(
  {
    slug: { type: String, required: true, unique: true, lowercase: true },
    name: { type: String, required: true, maxlength: 100 },
    description: { type: String, maxlength: 500 },
    iconUrl: { type: String },
    bannerUrl: { type: String },
    createdBy: { type: Schema.Types.ObjectId, ref: "User" },
    stats: {
      memberCount: { type: Number, default: 0 },
      postCount: { type: Number, default: 0 },
    },
  },
  { timestamps: true },
);

export type IOrbit = InferSchemaType<typeof OrbitSchema> & {
  _id: Types.ObjectId;
};

export const Orbit = getModel<IOrbit>("Orbit", OrbitSchema);
